src/txn/txn.c

```diff
@@ -895,10 +966,15 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
      * will be only one uncommitted prepared update. There can be a false positive of fixing history
      * store when handling prepared inserts, but it doesn't cost much.
      */
-    if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY) && resolved && upd == NULL)
-        WT_RET_NOTFOUND_OK(__txn_fixup_prepared_update(session, op, *cursorp, commit));
+    if (fix_upd != NULL)
+        WT_ERR(__txn_fixup_prepared_update(session, hs_cursor, fix_upd, commit));
 
-    return (0);
+err:
+    if (hs_cursor != NULL)
+        ret = __wt_hs_cursor_close(session, session_flags, is_owner);
+    if (commit)
+        __wt_free(session, fix_upd);
```

Comment:
@kommiharibabu This makes sense. But I wonder if we can do better than this.

Reply:
e680676